### PR TITLE
IPMIView: 2.17.0 -> 2.18.0

### PR DESCRIPTION
**Pull request by iamdenchik**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by vlaci — CHANGES_REQUESTED** [review]
The package still doesn't build for me because the hash is not correct and the patchelf issue needs to be resolved. See my comments in https://github.com/NixOS/nixpkgs/pull/106164. It's okay if you don't know how to fix the libIKVM patching issue by yourself, I can help you if you'd like.

You can try building the derivation yourself via issueing the following command in the root of your nixpkgs git checkout:
```terminal
$ nix-build . -A ipmiview
these derivations will be built:
  /nix/store/lig12cvpbjmm74h4nkid9brhk2fq7bgx-IPMIView_2.18.0_build.201007_bundleJRE_Linux_x64.tar.gz.drv
  /nix/store/wffdmz8m03552z4625f0r0xk04hq7s3h-IPMIView-2.18.0.drv
these paths will be fetched (0.00 MiB download, 0.01 MiB unpacked):
  /nix/store/1l18i82158dndfx7mdg6wxqjb476lfqg-mirrors-list
copying path '/nix/store/1l18i82158dndfx7mdg6wxqjb476lfqg-mirrors-list' from 'https://cache.nixos.org'...
building '/nix/store/lig12cvpbjmm74h4nkid9brhk2fq7bgx-IPMIView_2.18.0_build.201007_bundleJRE_Linux_x64.tar.gz.drv'...

trying https://www.supermicro.com/wftp/utility/IPMIView/Linux/IPMIView_2.18.0_build.201007_bundleJRE_Linux_x64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  103M  100  103M    0     0  9475k      0  0:00:11  0:00:11 --:--:-- 11.0M
hash mismatch in fixed-output derivation '/nix/store/4mx1wp32ijww8w33znjxnvrv33wr1kz9-IPMIView_2.18.0_build.201007_bundleJRE_Linux_x64.tar.gz':
  wanted: sha256:14cl9jhp2wvv5dgv4akkp75nqjz21f6j9pi6743s3rp298rpj095
  got:    sha256:10cv63yhh81gjxahsg4y3zp4mjivc217m4z1vcpwvvnds46c65h8
cannot build derivation '/nix/store/wffdmz8m03552z4625f0r0xk04hq7s3h-IPMIView-2.18.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/wffdmz8m03552z4625f0r0xk04hq7s3h-IPMIView-2.18.0.drv' failed
```

**Comment by iamdenchik** [comment]
> It's okay if you don't know how to fix the libIKVM patching issue by yourself, I can help you if you'd like.

help me, please)

```
Running desktop-file validation
building '/nix/store/xawpdj7id36y11vzqp2pid2f599wqawy-IPMIView_2.18.0_build.201007_bundleJRE_Linux_x64.tar.gz.drv'...

trying https://www.supermicro.com/wftp/utility/IPMIView/Linux/IPMIView_2.18.0_build.201007_bundleJRE_Linux_x64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  103M  100  103M    0     0  7699k      0  0:00:13  0:00:13 --:--:-- 7679k
copying path '/nix/store/a1hpnpjrxdllbfg179vzj7hnh581sz3c-gcc-wrapper-9.3.0' from 'https://cache.nixos.org'...
copying path '/nix/store/qkgbmfnkxw51sfzrxlxsbj6kf2zi7ab2-stdenv-linux' from 'https://cache.nixos.org'...
building '/nix/store/vrc3a2bp9v4np94gjazwwr98bwxbw7lz-IPMIView-2.18.0.drv'...
unpacking sources
unpacking source archive /nix/store/4mx1wp32ijww8w33znjxnvrv33wr1kz9-IPMIView_2.18.0_build.201007_bundleJRE_Linux_x64.tar.gz
source root is IPMIView_2.18.0_build.201007_bundleJRE_Linux_x64
setting SOURCE_DATE_EPOCH to timestamp 1602065497 of file IPMIView_2.18.0_build.201007_bundleJRE_Linux_x64/libiKVM_v11_64.so
patching sources
configuring
no configure script, doing nothing
building
patchelf: no section headers. The input file is probably a statically linked, self-decompressing binary
builder for '/nix/store/vrc3a2bp9v4np94gjazwwr98bwxbw7lz-IPMIView-2.18.0.drv' failed with exit code 1
error: build of '/nix/store/vrc3a2bp9v4np94gjazwwr98bwxbw7lz-IPMIView-2.18.0.drv' failed
```

**Comment by vlaci** [comment]
So the issue is with the `patchElf` call at https://github.com/NixOS/nixpkgs/blob/2d89106346fdf2801cd0419b7f52c47d1db18212/pkgs/applications/misc/ipmiview/default.nix#L35
I don't know what changed in IPMIView which casues it to fail but have a workaround suggestion:

1.  Remove this and the immediately above calls. This makes the derivation to build but the KVM console functionality is broken.
2. To fix the KVM console via an other method, you can extend the `LD_LIBRARY_PATH` environment variable with the path to `gcc-unwrapped.lib` when `IPMIView is started`. To do that, we need to have a `makeWrapper` call. Fortunately there is an existing `makeWrapper` invocation here: https://github.com/NixOS/nixpkgs/blob/2d89106346fdf2801cd0419b7f52c47d1db18212/pkgs/applications/misc/ipmiview/default.nix#L58-L59
Adding `gcc-unwrapped.lib` after `fontconfig` seems to make KVM console working again. 

<details>
<summary>Here is the complete diff I suggest (click to expand)</summary>

```diff
@@ -31,8 +31,6 @@ stdenv.mkDerivation rec {
   ''
     patchelf --set-rpath "${stdenv.lib.makeLibraryPath [ libX11 libXext libXrender libXtst libXi ]}" ./jre/lib/amd64/libawt_xawt.so
     patchelf --set-rpath "${stdenv.lib.makeLibraryPath [ freetype ]}" ./jre/lib/amd64/libfontmanager.so
-    patchelf --set-rpath "${gcc-unwrapped.lib}/lib" ./libiKVM64.so
-    patchelf --set-rpath "${gcc-unwrapped.lib}/lib" ./libiKVM_v11_64.so
     patchelf --set-rpath "${gcc.cc}/lib:$out/jre/lib/amd64/jli" --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" ./jre/bin/java
     patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" ./BMCSecurity/${stunnelBinary}
   '';
@@ -56,7 +54,7 @@ stdenv.mkDerivation rec {
     # WORK_DIR: unfortunately the ikvm related binaries are loaded from
     #           and user configuration is written to files in the CWD
     makeWrapper $out/jre/bin/java $out/bin/IPMIView \
-      --set LD_LIBRARY_PATH "${stdenv.lib.makeLibraryPath [ fontconfig ]}" \
+      --set LD_LIBRARY_PATH "${stdenv.lib.makeLibraryPath [ fontconfig gcc-unwrapped.lib ]}" \
       --prefix PATH : "$out/jre/bin:${iputils}/bin:${psmisc}/bin" \
       --add-flags "-jar $out/IPMIView20.jar" \
       --run 'WORK_DIR=''${XDG_DATA_HOME:-~/.local/share}/ipmiview
```

</details>

**Comment by iamdenchik** [comment]
@vlaci thx man! build succesed

File: pkgs/applications/misc/ipmiview/default.nix [hunk 1]
```diff
@@ -18,7 +18,7 @@ stdenv.mkDerivation rec {
 
   src = fetchurl {
     url = "https://www.supermicro.com/wftp/utility/IPMIView/Linux/IPMIView_${version}_build.${buildVersion}_bundleJRE_Linux_x64.tar.gz";
-    sha256 = "14cl9jhp2wvv5dgv4akkp75nqjz21f6j9pi6743s3rp298rpj095";
```
Comment (vlaci) [inline pkgs/applications/misc/ipmiview/default.nix]: Could you move this hash change in the previous commit where the version is bumped? I am also OK with squashing the two commits together.

**Review by vlaci — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/ipmiview/default.nix (hunk 1 above)
Comment (iamdenchik) [reply pkgs/applications/misc/ipmiview/default.nix]: done

**Comment by vlaci** [comment]
Just one nitpick, please keep the original commit message header `IPMIView: 2.17.0 -> 2.18.0`.
The contribution guide contains further information on commit messages https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

**Comment by iamdenchik** [comment]
> Just one nitpick, please keep the original commit message header `IPMIView: 2.17.0 -> 2.18.0`.
> The contribution guide contains further information on commit messages https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

fixed

**Review by vlaci — APPROVED** [review]

**Comment by vlaci** [comment]
LGTM

**Comment by iamdenchik** [comment]
> 
> 
> LGTM
> 
> Thank you, for the contribution!

thank you to you!

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 106428` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ipmiview</li>
  </ul>
</details>

**Review by vlaci — APPROVED** [review]

**Comment by iamdenchik** [comment]
@vlaci  Hello!

when will this task be merged to master?

**Comment by vlaci** [comment]
As soon as a maintainer with write access picks it up. I can review packages I maintain but I am not able to merge.

**Comment by iamdenchik** [comment]
> As soon as a maintainer with write access picks it up. I can review packages I maintain but I am not able to merge.

got it

**Comment by iamdenchik** [comment]
@mweinelt pls, approve

**Review by mweinelt — APPROVED** [review]
Tested by logging into an X7SPA board, checking sensors, opening KVM console. LGTM

**Comment by iamdenchik** [comment]
@mweinelt Hello!

When merge this commit to nixpkgs-unstable? 

I have to create PR for merge?

**Comment by vlaci** [comment]
The nixpkgs manual [contains some information](https://nixos.org/manual/nixpkgs/stable/#overview-of-nixpkgs) how and when the channels are updated. TL; DR nixpkgs-unstable is usually updated in every few days. No need for additional PR.

**Comment by iamdenchik** [comment]
> The nixpkgs manual [contains some information](https://nixos.org/manual/nixpkgs/stable/#overview-of-nixpkgs) how and when the channels are updated. TL; DR nixpkgs-unstable is usually updated in every few days. No need for additional PR.

thx!

**Pull request merged**